vms/platformvm/state.go

```diff
@@ -389,25 +389,28 @@ func (vm *VM) removeUTXO(db database.Database, utxoID ids.ID) error {
 // Only returns UTXOs after [start].
 // Returns at most [limit] UTXO IDs.
 // Returns nil if no UTXOs reference [addr].
-func (vm *VM) getReferencingUTXOs(db database.Database, addr []byte, start ids.ID, limit int) (ids.Set, error) {
-	toFetch := limit
-	utxoIDs := ids.Set{}
+func (vm *VM) getReferencingUTXOs(db database.Database, addr []byte, start ids.ID, limit int) ([]ids.ID, error) {
+
+	idSlice := []ids.ID(nil)
```

Comment:
Should we pre-allocate this array to `limit` to avoid multiple memory allocations? Or do we think that could be too space in-efficient with smaller calls?

Reply:
I think we're better off biting the bullet of increasing the slice. I'm guessing this a method that's used more in frequency than in size.